Give the example for custom magics in the docs.

Carreau opened the pull request:
The docs are registering magics by getting the current instance of
IPython and registering directly against it. A better approach is to
actually show the example on how to use the `load_ipython_extension`
function to correctly define a magic.

Carreau committed:
Give the example for custom magics in the docs.

The docs are registering magics by getting the current instance of
IPython and registering directly against it. A better approach is to
actually show the example on how to use the `load_ipython_extension`
function to correctly define a magic.

takluyver commented on docs/source/config/custommagics.rst:
```diff
@@ -84,12 +84,17 @@ IPython object:
+    def load_ipython_extension(ipython):
+        """
+        Any nodule file that define a function named load_ipython_extension can
+        be loaded via `load_ext` or be configured to be autoloaded by IPython at
```
Let's put `%load_ext` to make it clearer that it's a magic.

takluyver commented on docs/source/config/custommagics.rst:
```diff
@@ -108,11 +113,16 @@ instantiate the class yourself before registration:
+        """
+        # This class must then be registered with a manually created instance,
+        # since its constructor has different arguments from the default:
+        magics = StatefulMagics(ip, some_data)
```
ip -> ipython as that's what it's called in the function definition.

Carreau commented:
Feedback from sprints, it's unclear that load_ipython_extension should be in `__init__.py` and maybe link to flit on how to publish a package.

Carreau committed:
Fixes, plus complete example.

Also link to flit, as during JupyterCon, there was good feedback that it
looked great and should probably be mentioned in the magics docs.

Carreau commented:
Thanks !

On Sat, Sep 2, 2017 at 5:22 AM, Thomas Kluyver <notifications@github.com>
wrote:

> Merged #10750 <https://github.com/ipython/ipython/pull/10750>.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ipython/ipython/pull/10750#event-1232320235>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAUez1Zj_7OQhc5i7P4EFyUpwU8xSQyrks5seUh_gaJpZM4O8AMz>
> .
>
